### PR TITLE
Remove the druid_coordinator_metadata feature flag.

### PR DESCRIPTION
**Pull request by archolewa**
-- This feature flag no longer makes any sense, because customers must
either:
    1. Write a custom `PhysicalTable` with an Availability service
    that does not rely on the DataSourceMetadataService, in which case
    the feature flag has zero impact.
    2. Turn on the feature flag, so that availability gets
    populated.

Otherwise, the availability information will never get populated, and
every data request will fail with an error about how the metadata
service doesn't know anything about the specified datasource, a message
that can be a little bit misleading.

In other words, vanilla Fili doesn't work if that feature flag isn't
turned on, so there's no point in including it.

### CODE REVIEW

**Comment by archolewa** [comment]
@garyluoex There isn't really any point in deprecating it when it straight up breaks things. Typically you deprecate something that still works, but may be removed at some point in the future. This just straight up doesn't work.

**Comment by archolewa** [comment]
Also, I'm not sure what you mean by a "binding conflict" So far as I know, I'm not doing any binding here.

**Comment by archolewa** [comment]
Oh you mean the part where we're pulling binding on the metadata webservice out of the if-block. Sure, bindings can have conflicts, but we won't be able to compile if there is, and the whole idea of a named binding is how you resolve most such conflicts (give two conflicting bindings different names).

File: fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java [hunk 1]
```diff
@@ -137,6 +134,8 @@
 
 import javax.validation.constraints.NotNull;
 
+import rx.subjects.PublishSubject;
```
Comment (tarrantzhang) [inline fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java]: The order of this import probably should not change.
